File: app/src/main/java/com/plbear/iweight/base/BaseActivity.java
package com.plbear.iweight.base;

import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.plbear.iweight.utils.LogInfo;

import butterknife.ButterKnife;

/**
 * Created by yanyongjun on 2018/6/11.
 */

public abstract class BaseActivity extends AppCompatActivity {
    public abstract int getLayout();

    public abstract void afterLayout();

    protected static String TAG = BaseActivity.class.getSimpleName();

    protected Handler mHandler = new Handler();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        int layout = getLayout();
        if (layout != 0) {
            setContentView(layout);
            ButterKnife.bind(this);
        }
        loginfo("onCreate");
        AppManager.getAppManager().addToStack(this);
        afterLayout();
    }

    @Override
    protected void onStart() {
        super.onStart();
        loginfo("onStart");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        AppManager.getAppManager().removeFromStack(this);
        mHandler.removeCallbacksAndMessages(null);
        loginfo("onDestory");
    }

    @Override
    protected void onResume() {
        super.onResume();
        loginfo("onResume");
    }

    protected void loginfo(String info) {
        LogInfo.i(TAG, info);
    }

    protected void logerror(String error) {
        LogInfo.e(TAG, error);
    }

    public void exitAll() {
        AppManager.getAppManager().finishAllActivity();
    }

    @Override
    public void finish() {
        AppManager.getAppManager().removeFromStack(this);
        super.finish();
    }

    public void onClick_back(View v){
        loginfo("onCick_back");
        finish();
    }

}

File: app/src/main/java/com/plbear/iweight/base/BaseActivity.kt
package com.plbear.iweight.base

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.plbear.iweight.utils.LogInfo
import com.plbear.iweight.utils.MyLog

/**
 * Created by yanyongjun on 2018/normal_1/28.
 */
abstract class BaseActivity : AppCompatActivity() {
    abstract fun getLayout(): Int
    abstract fun afterLayout()
    protected val TAG = this.javaClass.simpleName

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        var layout = getLayout()
        if (layout != 0) {
            setContentView(getLayout())
        }
        MyLog.d(TAG, "onCreate")
        AppManager.getAppManager().addToStack(this)
        afterLayout()
    }


    override fun onDestroy() {
        super.onDestroy()
        AppManager.getAppManager().removeFromStack(this)
        MyLog.d(TAG, "onDestroy")
    }

    override fun onStart() {
        super.onStart()
        MyLog.d(TAG, "onStart")
    }

    override fun onResume() {
        super.onResume()
        MyLog.d(TAG, "onResume")
    }

    protected fun exitAll() {
        AppManager.getAppManager().finshAllActivity()
    }

    override fun finish() {
        AppManager.getAppManager().removeFromStack(this)
        super.finish()
    }

    fun loginfo(info: String) {
        LogInfo.e(TAG, info)
    }

    fun logerror(error: String) {
        LogInfo.e(TAG, error)
    }
}